lib/comm/ccp_message_handler.c

```diff
@@ -53,8 +53,18 @@ static void ccp_handle_create_alarm(char *message)
 {
     // extract data from message
     request server_request;
-    ccp_parse_request(message, &server_request);
+    CCP_PARSING_STATUS parsing_status = ccp_parse_request(message, &server_request);
 
+    if (parsing_status == CCP_PARSING_INVALID_WRONG_FORMAT)
+    {
+        ccp_message_sender_send_response(server_request.action_type, CCP_STATUS_BAD_REQUEST, "Invalid fromat of the request.");
+        return;
+    }
+    else if (parsing_status == CCP_PARSING_INVALID_EMPTY_POINTER)
+    {
+        ccp_message_sender_send_response(server_request.action_type, CCP_STATUS_SERVER_ERROR, "Server error occurred.");
+        return;
+    }
```

Comment:
Having a static function would improve the DRY

Reply:
good stuff, refactored to make it a separate func